iothub_client/tests/common_dt_e2e/iothubclient_common_dt_e2e.c

```diff
@@ -38,885 +36,943 @@
-            Unlock(device->lock);
+            Unlock(device_desired_data->lock);
         }
         ThreadAPI_Sleep(1000);
     }
 
-    ASSERT_IS_TRUE(difftime(nowTime, beginOperation) < MAX_CLOUD_TRAVEL_TIME, "Timeout waiting for twin message");
+    ASSERT_IS_TRUE(difftime(now_time, begin_operation) < MAX_CLOUD_TRAVEL_TIME, "Timeout waiting for twin message");
 
-    // unsubscribe
-    setdevicetwincallback_on_device_or_module(NULL, NULL);
+    // Unsubscribe
+    _set_device_twin_callback(NULL, NULL);
 
-    if (Lock(device->lock) != LOCK_OK)
+    // Check results.
+    if (Lock(device_desired_data->lock) != LOCK_OK)
     {
         ASSERT_FAIL("Lock failed");
     }
     else
     {
-        ASSERT_IS_NOT_NULL(root_value, "json_parse_string failed");
         ASSERT_ARE_EQUAL(char_ptr, expected_desired_string, string_property, "string data retrieved differs from expected");
         ASSERT_ARE_EQUAL(int, expected_desired_integer, integer_property, "integer data retrieved differs from expected");
         ASSERT_ARE_EQUAL(char_ptr, expected_desired_string, string_property_from_array, "string data (from array) retrieved differs from expected");
         ASSERT_ARE_EQUAL(int, expected_desired_integer, integer_property_from_array, "integer data (from array) retrieved differs from expected");
 
-        (void)Unlock(device->lock);
-
-        // cleanup
-        json_value_free(root_value);
-        IoTHubDeviceTwin_Destroy(serviceClientDeviceTwinHandle);
-        IoTHubServiceClientAuth_Destroy(iotHubServiceClientHandle);
-        free(expected_desired_string);
-        free(buffer);
-        destroy_on_device_or_module();
-        device_desired_deinit(device);
+        (void)Unlock(device_desired_data->lock);
     }
+
+    // Cleanup
+    free(string_property_from_array);
+    free(string_property);
+    free(buffer);
+    free(expected_desired_string);
+    free(twin_data);
+    IoTHubDeviceTwin_Destroy(serviceclient_devicetwin_handle);
+    IoTHubServiceClientAuth_Destroy(iothub_serviceclient_handle);
+    _device_desired_data_deinit(device_desired_data);
+    _breakdown_test();
 }
 
-// request_full_twin_and_wait_for_response requests the full twin from the Hub (using device, NOT service sdk).
-// Verifies that the server sends a response or else times out.
-static void request_full_twin_and_wait_for_response(IOTHUB_PROVISIONED_DEVICE* deviceToUse, DEVICE_DESIRED_DATA *deviceDesiredData)
+void _client_create_with_properties_and_send_d2c(IOTHUB_PROVISIONED_DEVICE* device_to_use, MAP_HANDLE mapHandle)
 {
-    if (deviceToUse->moduleConnectionString != NULL)
-    {
-        ASSERT_ARE_EQUAL(IOTHUB_CLIENT_RESULT, IoTHubModuleClient_GetTwinAsync(iothub_moduleclient_handle, deviceTwinCallback, deviceDesiredData), IOTHUB_CLIENT_OK);
-    }
-    else
+    IOTHUB_MESSAGE_HANDLE msgHandle;
+
+    char messageStr[512];
+    int len = snprintf(messageStr, sizeof(messageStr), "Happy little message from device '%s'", device_to_use->deviceId);
```

Comment:
:)

Reply:
lolol some sections in the refactor I left alone, but because so much was moved around it looks new.  I'm wondering if this is that!